### PR TITLE
Feature/multi interpolation

### PR DESCRIPTION
**Pull request by whaeck**
This adds the InterpolationTable object. It represents a general interpolation table.

The various other Table objects were extended to allow for a template specification of the underlying x and y containers (this allows us to use IteratorView instead of std::vector). This was useful for using it under the hood in InterpolationTable without having to split up the table's x and y vector.

### CODE REVIEW

File: src/scion/math/InterpolationTable/src/ctor.hpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  new (this) InterpolationTable( base );
+  return *this;
+}
+
+/**
+ *  @brief Move assignment using a continuous energy table
+ *
+ *  @param[in] table    the continuous energy table to be moved
+ */
+InterpolationTable& operator=( InterpolationTable&& base ) {
```
Comment (BobbiRiedel) [inline src/scion/math/InterpolationTable/src/ctor.hpp:43]: Copying yourself can be problematic, you may be copying and assigning yourself.

File: src/scion/math/LinearLogTable.hpp
Comment (BobbiRiedel) [inline src/scion/math/LinearLogTable.hpp:1]: Why no baseclass????

**Review by BobbiRiedel — COMMENTED** [review]
Made a couple comments.

File: src/scion/math/InterpolationTable/src/ctor.hpp (hunk 1 above)
Comment (whaeck) [reply src/scion/math/InterpolationTable/src/ctor.hpp:43]: Yes, I should fix that at some point. I had noticed this one already but I did not fix it yet.

File: src/scion/math/LinearLogTable.hpp
Comment (whaeck) [reply src/scion/math/LinearLogTable.hpp:1]: it will happen. Be patient, my young padawan.

Comment (whaeck) [reply src/scion/math/LinearLogTable.hpp:1]: It was implemented in #47

**Review by BobbiRiedel — APPROVED** [review]

**Pull request merged**